src/main.rs

```diff
@@ -153,16 +153,30 @@ enum Sevctl {
     },
 }
 
-fn download(
-    rsp: reqwest::Result<reqwest::blocking::Response>,
-    usage: Usage,
-) -> Result<sev::Certificate> {
-    let rsp = rsp.context(format!("unable to contact {} server", usage))?;
-
-    let status = rsp.status();
-    let mut rsp = rsp.error_for_status().context(&format!(
-        "received failure from {} server: {}",
-        usage, status
+use std::time::Duration;
+
+fn download(url: &str, usage: Usage) -> Result<sev::Certificate> {
+    let mut rsp = reqwest::blocking::get(url);
+    for request_wait_seconds in &[0, 2, 4, 6, 9] {
+        std::thread::sleep(Duration::from_secs(*request_wait_seconds));
+        match &rsp {
+            // Connection is established, ensure that status code is not an error.
+            Ok(found) => {
+                if found.status().is_success() {
+                    break;
+                } else {
+                    rsp = reqwest::blocking::get(url);
+                }
+            }
+            // Connection not established, retry.
+            Err(_) => {
+                rsp = reqwest::blocking::get(url);
+            }
+        }
+    }
+    let mut rsp = rsp.context(format!(
+        "After 5 attempts, unable to contact {} server",
+        usage
```

Comment:
This isn't very actionable for the user. Furthermore, if we update the number of attempts we'll have to remember to update this message. I think since we are potentially retrying multiple times and we may get different non-200 HTTP replies, it might be a good idea to have a Vec<String> that we push the HTTP reply's error message to each time we decide we have to retry. Then we could include this information which the end user might find helpful if they need to file a bug report.

I.e., we could change this to:

```rust
let mut rsp = rsp.context("contacting {} server failed:\n{}", usage, retried_replies.join("\n"))?;
```